cpp/dolfinx/generation/RectangleMesh.cpp

```diff
@@ -25,7 +24,7 @@ mesh::Mesh build_tri(MPI_Comm comm,
                      std::array<std::size_t, 2> n,
                      const mesh::GhostMode ghost_mode,
                      const mesh::CellPartitionFunction& partitioner,
-                     const std::string& diagonal)
+                     const generation::DiagonalType diagonal)
```

Comment:
In the DOLFINx style we don't normally use `const` when passing enums.

Reply:
We use it for GhostMode in this function. Do you want me to change that as well?